webpack/assets/javascripts/react_app/components/common/forms/__tests__/InputFactory.test.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { shallow } from 'enzyme';
+import toJson from 'enzyme-to-json';
+import { testComponentSnapshotsWithFixtures } from '../../../../common/testHelpers';
+import {
+  ownComponentFieldProps,
+  formAutocompleteDataProps,
+} from '../FormField.fixtures';
+import InputFactory, { registerInputComponent } from '../InputFactory';
+
+const fixtures = {
+  'renders text input': { type: 'text', name: 'a' },
+  'renders DateTime input': { type: 'dateTime', name: 'a' },
+  'renders AutoComplete': {
+    type: 'autocomplete',
+    ...formAutocompleteDataProps,
+  },
+};
+
+const Abc = props => <input type="hidden" id={props.id} name={props.name} />;
+Abc.propTypes = {
+  id: PropTypes.string,
+  name: PropTypes.string,
+};
+Abc.defaultProps = {
+  id: undefined,
+  name: null,
+};
+
+describe('InputFactory', () => {
```

Comment:
The testing approach makes sense to me.

Could you make it simpler by using the `testComponentSnapshotsWithFixtures` for the own component test?
Maybe by doing:
```js
  describe('rendering', () => {
    registerInputComponent('ownInput', Abc);
    testComponentSnapshotsWithFixtures(InputFactory, fixtures);
  });
```